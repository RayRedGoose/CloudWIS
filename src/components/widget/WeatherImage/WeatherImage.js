import React from 'react'
import './WeatherImage.scss'
import RainyBackground from '../backgrounds/rainy/RainyBackground/RainyBackground'
import SnowyBackground from '../backgrounds/snowy/SnowyBackground/SnowyBackground'

const WeatherImage = ({type}) => {
  const findBackground = (type) => {
    if (type === 'rain') {
      return (<RainyBackground />)
    }

    if (type === 'snow') {
      return (<SnowyBackground />)
    }

    return (<div></div>)
  }

  return (
    <section className="weather-image">
      { findBackground(type) }
    </section>
  )
}

export default WeatherImage
