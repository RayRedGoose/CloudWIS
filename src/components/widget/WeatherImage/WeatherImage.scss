.weather-image {
  background-color: #fff;
  background-image: url('../backgrounds/images/bg.svg');
  background-position: center;
  background-size: cover;
  border-radius: 20px;
  width: 30vw; height: 100%;
  overflow: hidden;
}
