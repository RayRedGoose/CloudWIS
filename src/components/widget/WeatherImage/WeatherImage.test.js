import React from 'react';
import { shallow } from 'enzyme';
import WeatherImage from './WeatherImage';

it('should match the snapshot with all data passed in correctly', () => {
  const block = shallow(
    <WeatherImage type={'clouds'} />
  );

  expect(block).toMatchSnapshot();
});

it('should match the snapshot with rain data passed', () => {
  const block = shallow(
    <WeatherImage type={'rain'} />
  );

  expect(block).toMatchSnapshot();
});

it('should match the snapshot with snow data passed', () => {
  const block = shallow(
    <WeatherImage type={'snow'} />
  );

  expect(block).toMatchSnapshot();
});
