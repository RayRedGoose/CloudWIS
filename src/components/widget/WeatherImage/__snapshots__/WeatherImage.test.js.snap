// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot with all data passed in correctly 1`] = `
<section
  className="weather-image"
>
  <div />
</section>
`;

exports[`should match the snapshot with rain data passed 1`] = `
<section
  className="weather-image"
>
  <RainyBackground />
</section>
`;

exports[`should match the snapshot with snow data passed 1`] = `
<section
  className="weather-image"
>
  <SnowyBackground />
</section>
`;
